compiler/src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -134,13 +134,17 @@ class Erasure extends Phase with DenotTransformer {
   }
 
   def assertErased(tp: Type, tree: tpd.Tree = tpd.EmptyTree)(implicit ctx: Context): Unit = {
-    def isAllowed(cls: Symbol, sourceName: String) =
-      tp.typeSymbol == cls && ctx.compilationUnit.source.file.name == sourceName
```

Comment:
I think you can just replace `tp.typeSymbol` with `tp.widen.typeSymbol` here and otherwise keep the old definition of `isAllowed`.